src/Swarm/Game/Step.hs

```diff
@@ -1240,10 +1240,47 @@ execConst c vs s k = do
         inv <- use robotInventory
         return $ Out (VInt (fromIntegral $ countByName name inv)) s k
       _ -> badConst
+    Scout -> case vs of
+      [VDir d] -> do
+        rMap <- use robotMap
+        myLoc <- use robotLocation
+        heading <- deriveHeading d
+        botsByLocs <- use robotsByLocation
+        selfRid <- use robotID
+
+        -- Includes the base location, so we exclude the base robot later.
+        let locsInDirection :: [Location]
+            locsInDirection = take maxScoutRange $ iterate (.+^ heading) myLoc
```

Comment:
```suggestion
            locsInDirection = take maxScoutRange . drop 1 $ iterate (.+^ heading) myLoc
```
Would it be easier to do this, so we don't include the base location and don't have to exclude it later?

Reply:
I actually started with that version, but realized that I want to be able to detect a robot on the same cell as the base.